scim2-sdk-common/src/main/java/com/unboundid/scim2/common/BaseScimResource.java

```diff
@@ -253,18 +286,13 @@ protected Map<String, Object> getAny()
   }
 
   /**
-   * Adds the urn of this class to the list of schemas for this object.
-   * This is taken from the schema annotation of a class that extends
-   * this class.  If the class has no schema annotation, no schema urn
-   * will be added.
+   * Adds the URN of this class to the list of schemas for this object. This is
+   * taken from the {@link Schema} annotation of the subclass.
    */
-  private void addMyUrn()
+  private void addClassDefinedUrn()
   {
-    String mySchema = SchemaUtils.getSchemaUrn(this.getClass());
-    if ((mySchema != null) && (!mySchema.isEmpty()))
```

Comment:
`getSchemaUrn()` never returns a null or empty string, so I've removed this unneccessary check.